.github/workflows/go.yml

```diff
@@ -0,0 +1,58 @@
+name: Go
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write  # Enable OIDC
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.18
+
+    - name: Build
+      run: go build -v ./...
+
+    - name: Unit Tests
+      run: go test -v ./...
+      
+    - name: Test Sign commit
+      run: |
+        set -e
+
+        # Setup repo + tool
+        go install .
+        export PATH="$PATH:$GOPATH/bin"
+        echo "PATH=${PATH}"
+        whereis gitsign
+        mkdir /tmp/git
+        cd /tmp/git
+        git init -b main .
+        git config --global user.email "test@example.com"
+        git config --global user.name "gitsign"
+        git config --global gpg.x509.program gitsign
+        git config --global gpg.format x509
+        git config --global commit.gpgsign true
+
+        # Verify tool is on our path
+        gitsign -h
+
+        # Sign commit
```

Comment:
```suggestion
        # Sign and verify commit
```

Reply:
Want to keep diffs largely unchanged for this pass - will follow up with another PR for this!